### ISSUE TITLE
IdP attribute to MELLON environment variable mapping not working

### ISSUE THREAD

**Issue by itewk** [openshift/request-header-saml-service-provider#15]
https://github.com/openshift/request-header-saml-service-provider/blob/4a3af6e0c9e5f858aafebc2a572d6cca0f77682c/openshift.conf#L70-L72

While going through this with another cluster just today ran into an issue.

* IdP was sending the user name as `Username` attribute
* we set `REMOTE_USER_SAML_ATTRIBUTE` environment variable to `Username`
* Mellon_Username environment variable showing up in mellon debug logs
* Mellon_user environment variable NOT showing up in mellon debug logs
* the Remote-User header was NOT being set, presumably because MELLON_user was not being set

it would seem something is not working with https://github.com/openshift/request-header-saml-service-provider/blob/master/openshift.conf#L70. When we updated the IdP to send `User` as the user name attribute and updated the `REMOTE_USER_SAML_ATTRIBUTE` to be `User` as well it worked fine. we also tried `foobar` and that didn't work. It seems to only work when the IdP attribute is `User` which aligns with https://github.com/openshift/request-header-saml-service-provider/blob/master/openshift.conf#L72 but i thought the point of https://github.com/openshift/request-header-saml-service-provider/blob/master/openshift.conf#L70 was so that we could use any attribute.

We also tested this for `name` and `email` and had the exact same experience.

I am not sure if this is an issue with mod_auth_mellon or with how we are using MellonSetEnv

**Comment by itewk** [comment]
@brenton any ideas?

**Comment by brenton** [comment]
Are you 100% certain REMOTE_USER_SAML_ATTRIBUTE was being set correctly?  It could be that 'user' is the default in mod_auth_mellon and you were hitting this logic instead:
https://github.com/openshift/request-header-saml-service-provider/blob/master/openshift.conf#L61

You could munge the configs to confirm exactly which block you were hitting.  This stuff can be really fiddly.  I don't doubt there could be a minor problem with the syntax in our default config though. :(

**Comment by itewk** [comment]
@brenton i ams sure as can possible be. We checked the environment variables names with copy paste and find more then once. And, when changing just the value, from `Username` to `user` and updating the IdP to send `user` instead of `username` it started working. then we switched to trying `foo` both at the IdP and the value of REMOTE_USER_SAML_ATTRIBUTE and it didn't work. Same held true for the other variables. We discovered this because the email was the only header that was working, but also was the only one who's value matched as `email`. We tried changing that value to `bar` at both IdP and the value of `REMOTE_USER_EMAIL_SAML_ATTRIBUTE` and it stopped working too.
